_episodes/03-monitor-the-model.md

```diff
@@ -570,123 +535,24 @@ plt.ylabel("true sunshine hours")
-~~~
-{: .language-python} 
 
-Model training remains entirely unchanged:
-~~~
-history = model.fit(X_train, y_train,
-                    batch_size = 50,
-                    epochs = 1000,
-                    validation_data=(X_val, y_val),
-                    callbacks=[earlystopper],
-                    verbose = 2)
-~~~
-{: .language-python} 
-
-But when now doing predictions, things will look different.
-Let us do two predictions an compare the results.
-
-~~~
-y_test_predicted1 = model.predict(X_test)
-y_test_predicted2 = model.predict(X_test)
-
-y_test_predicted1[:10], y_test_predicted2[:10]
-~~~
-{: .language-python} 
-
-This should give two arrays with different float numbers.
-
-We can now compute predictions for a larger ensemble, say 100 random variations of the same model:
-~~~
-from tqdm.notebook import tqdm  # optional: to add progress bar
-
-n_ensemble = 100
-y_test_predicted_ensemble = np.zeros((X_test.shape[0], n_ensemble))
-
-for i in tqdm(range(n_ensemble)):  # or: for i in range(n_ensemble):
-    y_test_predicted_ensemble[:, i] = model.predict(X_test)[:,0]
-~~~
-{: .language-python} 
-
-This will give an array of predictions, 100 different predictions for each datapoint in `X_test`.
-We can inspect an example distribution, for instance by plotting a histrogram:
-
-~~~
-plt.hist(y_test_predicted_ensemble[0,:], rwidth=0.9)
-plt.xlabel("predicted sunshine hours")
-~~~
-{: .language-python} 
-
-![Output of plotting sample](../fig/03_monte_carlo_dropout_distribution_example.png)
-
-Instead of full distributions for every datapoint we might also just want to extract the mean and standard deviation.
-~~~
-y_test_predicted_mean = np.mean(y_test_predicted_ensemble, axis=1)
-y_test_predicted_std = np.std(y_test_predicted_ensemble, axis=1)
-~~~
-{: .language-python} 
-
-This can then be plotted again as a scatter plot, but now with added information on the model uncertainty.
-~~~
-plt.figure(figsize=(5, 5), dpi=100)
-plt.scatter(y_test_predicted_mean, y_test, s=40*y_test_predicted_std, 
-            c=y_test_predicted_std, alpha=0.5)
-plt.xlabel("predicted")
-plt.ylabel("true values")
-~~~
-{: .language-python} 
-![Output of plotting sample](../fig/03_scatter_plot_model_uncertainty.png)
+> # Create a similar scatter plot as above for a reasonable baseline
+>
+> What can we take as a baseline? 
+> Maybe the simplest prediction to make would be to say: Tomorrow we will have the same number of sunshine hours as today.
+> Let's compare to this.
+> 
+> > # Solution
+> > We can here just take the `BASEL_sunhine` column of our data, because this contains the sunshine hours from one day before what we have as a label.
+> > ~~~
+> > plt.figure(figsize=(5, 5), dpi=100)
+> > plt.scatter(X_test["BASEL_sunshine"], y_test, s=10, alpha=0.5)
+> > plt.xlabel("sunshine hours yesterday")
+> > plt.ylabel("true sunshine hours")
+> > ~~~
+> > {: .language-python} 
+> > 
+> > ![Output of plotting sample](../fig/03_regression_test_5_naive_baseline.png)
```

Comment:
```suggestion
> ## Exercise: Create a similar scatter plot as above for a reasonable baseline
>
> What can we take as a baseline? 
> Maybe the simplest prediction to make would be to say: Tomorrow we will have the same number of sunshine hours as today.
> Let's compare to this.
> 
> > ## Solution
> > We can here just take the `BASEL_sunhine` column of our data, because this contains the sunshine hours from one day before what we have as a label.
> > ~~~
> > plt.figure(figsize=(5, 5), dpi=100)
> > plt.scatter(X_test["BASEL_sunshine"], y_test, s=10, alpha=0.5)
> > plt.xlabel("sunshine hours yesterday")
> > plt.ylabel("true sunshine hours")
> > ~~~
> > {: .language-python} 
> > 
> > ![Output of plotting sample](../fig/03_regression_test_5_naive_baseline.png)
> {:.solution}
{:.challenge}
```